src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -71,24 +61,32 @@ public PersonCard(ReadOnlyPerson person, int displayedIndex) {
 
     /**
      * This method takes in the tagName, and returns the color associated with that tagName
-     * If the if the tag has no associated color, a random one from the list available will be given.
-     * Once the list runs out, all other tags will be grey.
-     *
+     * If the if the tag has no associated color, a unique random color will be assigned to the tag.
+=     *
      * @param tagName is the String name of the tag
      * @return the color associated to the tagName
      */
     public static String obtainTagColors(String tagName) {
         if (!currentTagColors.containsKey(tagName)) {
-            if (availableColorsLeft.size() != 0) {
-                Random rand = new Random();
-                int randIndex = rand.nextInt(availableColorsLeft.size());
-                assignedColor = availableColorsLeft.get(randIndex);
-
-                currentTagColors.put(tagName, assignedColor);
-                availableColorsLeft.remove(randIndex);
-            } else {
-                currentTagColors.put(tagName, "GRAY");
-            }
+            do {
+                Random random = new Random();
+                final float hue = random.nextFloat();
+                final float saturation = 0.65f + random.nextFloat()
+                        * (0.90f - 0.65f);
+                final float luminance = 0.60f + random.nextFloat()
+                        * (0.90f - 0.60f);
+
+                Color color = Color.getHSBColor(hue, saturation, luminance);
+
+                Formatter hexRepresentation = new Formatter(new StringBuffer("#"));
+                hexRepresentation.format("%02X", color.getRed());
+                hexRepresentation.format("%02X", color.getGreen());
+                hexRepresentation.format("%02X", color.getBlue());
+                assignedColor = hexRepresentation.toString();
+            } while (usedColors.contains(assignedColor));
+
+            usedColors.add(assignedColor);
+            currentTagColors.put(tagName, assignedColor);
```

Comment:
Is it possible to only generate colours according to a colour scheme that fits our theme and font?

Reply:
I set the random range of color such that the random color will not be too bright for our dark theme.